### PR TITLE
fix: allow selecting tests and files containing regexp special characters

### PR DESCRIPTION
**Pull request by Andarist**
This issue was quite annoying :p Jest interprets pattern as regexp so if we select a test name containing a special character this gets (currently) interpreted in a wrong way by jest. This fixes the problem

### CODE REVIEW

**Comment by SimenB** [comment]
Is this an issue in jest itself? Also, just for test name not file name?

**Comment by Andarist** [comment]
I'm going to answer based on my limited knowledge about jest

> Is this an issue in jest itself?

I don't think so, from what I understand what gets passed to that jest's callback could actually be a regexp~ pattern and would be interpreted as regexp by jest. We call it in here after selecting from the list though and this is already "resolved" string which we don't want to actually be interpreted as regexp (but it will, so we need to provide a regexp pattern with special characters escaped).

>  just for test name not file name?

When preparing this I thought about this briefly but concluded that file patterns contain seperator (`/`) which is a special regexp character so it has to be handled already by jest. But when I think about it now - the problem must be the same, but maybe separators are somehow special-cased by jest? Not sure - gonna investigate this in a moment. Thanks for mentioning this!

**Comment by Andarist** [comment]
Ok, I've found out that path separator is indeed special-cased by jest here - https://github.com/facebook/jest/blob/70bed722d6b68456ffcb7df70d4a9b59bcea3d8a/packages/jest-core/src/lib/update_global_config.ts#L34-L37

I've implemented this fix for selecting file names as well. It's past midnight right now and I'm not sure if I will have time to write tests for this tomorrow, but I'd appreciate a code review on the logic when you get time. Maybe something else will come to your mind while I will be working on tests.

File: yarn.lock [hunk 1]
```diff
@@ -2876,7 +2876,7 @@ jest-pnp-resolver@^1.2.1:
 
 jest-regex-util@^24.3.0, jest-regex-util@^24.9.0:
   version "24.9.0"
-  resolved "https://registry.yarnpkg.com/jest-regex-util/-/jest-regex-util-24.9.0.tgz#c13fb3380bde22bf6575432c493ea8fe37965636"
+  resolved "https://registry.npmjs.org/jest-regex-util/-/jest-regex-util-24.9.0.tgz#c13fb3380bde22bf6575432c493ea8fe37965636"
```
Comment (SimenB) [inline yarn.lock]: Wat?

Comment (Andarist) [reply yarn.lock]: Seems like I've set the registry to npm after having some problems when installing things long time ago. I myself have even forgotten about this and noone has commented on this during my PRs 🤷‍♂ I recall yarn even recommending to set this in some issues as that yarn registry is just a proxy to npm or something. Dunno though - gonna just change my global config back to default and fix this here.

File: src/test_utils/pluginTester.js [hunk 2]
```diff
@@ -24,10 +24,13 @@ expect.addSnapshotSerializer({
   print: val => stripAnsi(val.replace(WINDOWS_CLEAR, '[MOCK - clear]')),
 });
 
-const pluginTester = (Plugin, config = {}) => {
-  const stdout = { columns: 80, write: jest.fn() };
+const pluginTester = (Plugin, options = {}) => {
+  const stdout = {
+    columns: (options.stdout || {}).columns || 80,
```
Comment (Andarist) [inline src/test_utils/pluginTester.js]: optional chaining plz 😉

File: src/file_name_plugin/__tests__/plugin.test.js [hunk 3]
```diff
@@ -110,3 +139,32 @@ it('file matching is case insensitive', async () => {
...
+    ],
+  });
+  const runPromise = plugin.run({}, updateConfigAndRun);
+
+  type('t', 'r', 'i', 'm', 'm', KEYS.ARROW_DOWN, KEYS.ENTER);
+  await runPromise;
+
+  expect(updateConfigAndRun).toHaveBeenCalledWith({
+    mode: 'watch',
+    testPathPattern: 'ing\\.js',
```
Comment (Andarist) [inline src/file_name_plugin/__tests__/plugin.test.js]: I believe this should have included more - width is set to 40 after all. But this is a separate issue that could be investigated.

**Comment by Andarist** [comment]
@SimenB I've added more tests and made some minor changes - so a re-review would be appreciated.

**Comment by Andarist** [comment]
Sure, no problem.

**Comment by SimenB** [comment]
Published as 0.4.1

**Pull request merged**